src/openforms/submissions/report.py

```diff
@@ -15,6 +16,35 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass
+class CoSigner:
+    co_sign_data: dict
+
+    @property
+    def representation(self) -> str:
+        return self.co_sign_data.get("representation", "")
+
+    @property
+    def identifier(self) -> str:
+        return self.co_sign_data.get("identifier", "")
+
+    @property
+    def auth_attribute(self) -> str:
+        return self.co_sign_data.get("co_sign_auth_attribute", "")
+
+    def __str__(self):
+        if not self.identifier:
+            return _(
+                "{representation} (missing {auth_attribute} for co-signer)"
+            ).format(
+                representation=self.representation, auth_attribute=self.auth_attribute
+            )
+        return f"{self.representation} ({self.auth_attribute}: {self.identifier})"
+
+    def __repr__(self):
+        return f"CoSigner(co_sign_data={self.co_sign_data}, auth_attribute={self.auth_attribute})"
```

Comment:
these are typically generated by the dataclass and there should not be a need to override them.

If you need a custom/particular string representation, implement them as a custom method and call that in the template